types/V1.mli

```diff
@@ -190,6 +190,22 @@ module type FLOW = sig
       before returning. At this point no data can flow in either direction
       and resources associated with the flow can be freed.
       *)
+
+  val disconnect: flow -> unit io
```

Comment:
Why does this type need to be `unit io` and not just `unit`?  The substantive difference with `close` is that this does a dirty disconnect, and also the resources being freed asynchronously doesn't seem that bad.  The main objection to doing it asynchronously is that there won't be any global way to sync on (e.g.) not running out of file descriptors.  However, this global pool can be handled by the implementation of `type t` (that can track the total set of allocated fds), instead of individually at the `disconnect` function call level.

So I'm inclined to argue for `disconnect` to be a non-blocking `flow -> unit` function with asynchronous freeing of resources under all conditions.